CI: downstream tests failing due to environment resolvement (geospatial stack)

The geopandas downstream tests are failing because of an installation issue. 

See eg https://travis-ci.org/pandas-dev/pandas/jobs/508237658, giving: 

```
>   from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
E   ImportError: libiconv.so.2: cannot open shared object file: No such file or directory
```

This environment is defined by this yml file: https://github.com/pandas-dev/pandas/blob/master/ci/deps/travis-36.yaml

It is mixing defaults and conda-forge (with defaults as higher priority one), but given the `nomkl` it is getting blas / openblas from conda-forge. 
Until recently, the full geospatial stack was resolved to come from defaults (see eg https://travis-ci.org/pandas-dev/pandas/jobs/507830158#L1749), but since the last hours, it is getting fiona / gdal from conda-forge as well (but not all of the gdal dependencies, hence giving the error above).

cc @ocefpaf I know this is again an example of mixing conda-forge and defaults (like the recent one we had on the geopandas CI), which is not guaranteed to work. But it is also again something that was working until recently, and a kind of common use case: getting all from defaults, except for a few things that are not available there (and therefore adding conda-forge as the second, lower priority channel)

--- a/pandas/tests/test_downstream.py
+++ b/pandas/tests/test_downstream.py
@@ -102,10 +102,7 @@ def test_pandas_datareader():
 
 
 # importing from pandas, Cython import warning
-@pytest.mark.filterwarnings("ignore:The 'warn':DeprecationWarning")
-@pytest.mark.filterwarnings("ignore:pandas.util:DeprecationWarning")
 @pytest.mark.filterwarnings("ignore:can't resolve:ImportWarning")
-@pytest.mark.skip(reason="gh-25778: geopandas stack issue")
 def test_geopandas():
 
     geopandas = import_module("geopandas")  # noqa

